lib/diplomat/rest_client.rb

```diff
@@ -173,6 +170,88 @@ def use_consistency(options)
       options && options[:consistency] ? [options[:consistency].to_s] : []
     end
 
+    # rubocop:disable MethodLength, AbcSize, CyclomaticComplexity, PerceivedComplexity
+    # TODO: Migrate all custom params in options
+    def parse_options(options)
+      headers = nil
+      query_params = []
+      url_prefix = nil
+      consistency = []
+
+      headers = { 'X-Consul-Token': configuration.acl_token } if configuration.acl_token
+
+      unless options.nil?
```

Comment:
if options was a empty hash (by default) instead of nil, you could remove that condition

Reply:
+1

Reply:
I've set the default as empty hash and removed plenty of corresponding tests